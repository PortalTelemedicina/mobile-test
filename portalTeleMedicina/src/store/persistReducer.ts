import AsyncStorage from '@react-native-community/async-storage';
import { persistReducer } from 'redux-persist';

export default (reducers) => {
  const persistedReducer = persistReducer(
    {
      key: 'portalTelemedicina',
      storage: AsyncStorage,
      whitelist: [],
    },
    reducers
  );

  return persistedReducer;
};
