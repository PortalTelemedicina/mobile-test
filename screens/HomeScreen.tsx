import React, { useEffect } from 'react';
import { View, StyleSheet } from 'react-native';
import { useNetInfo } from '@react-native-community/netinfo';
import { useDispatch, useSelector } from 'react-redux';

/* System */
import { systemI18n, systemLayout, systemColors, systemConfig } from '../system';
import { RootTabScreenProps } from '../types';

/* Actions/Selectors */
import { getAllSpecialistCategoryRequest } from '../store/specialistCategory/specialistCategoryActions';
import {
  getAllSpecialistCategorySelector,
  getErrorSelector,
} from '../store/specialistCategory/specialistCategorySelectors';
import { isLoadingSelector } from '../store/system/systemSelectors';

/* Components */
import HomeMenuContainer from '../components/HomeMenuContainer';
import SpecialistCategoryList from '../components/SpecialistCategoryList';
import { TextContainer as Text } from '../components/TextContainer';
import ErrorContainer from '../components/ErrorContainer';
import SpecialistCategoryListShimmer from '../components/SpecialistCategoryListShimmer';

export default function HomeScreen({ navigation }: RootTabScreenProps<'HomeStackScreen'>): JSX.Element {
  const dispatch = useDispatch();
  const netInfo = useNetInfo();
  const isLoading = useSelector(isLoadingSelector);
  const specialistsCategories = useSelector(getAllSpecialistCategorySelector);
  const isError = useSelector(getErrorSelector);

  const getAllSpecialistCategory = () => {
    dispatch(getAllSpecialistCategoryRequest());
  };

  useEffect(() => {
    getAllSpecialistCategory();
  }, []);

  const isConnected = () => {
    console.log('IsConnect', netInfo.isConnected);
    return netInfo.isConnected;
  };

  const goToSpecialistListScreen = (category: string) => {
    navigation.push('SpecialistListScreen', { category });
  };

  const isFetched = () => {
    return specialistsCategories.length > 0;
  };

  return (
    <View style={styles.container}>
      <Text style={styles.welcome}>{systemI18n.t('label.hello')}</Text>
      <Text style={styles.username} bold>
        {systemConfig.getUserDefault()}
      </Text>
      <Text style={styles.subtitle} bold>
        {systemI18n.t('title.specialists')}
      </Text>
      <View style={styles.containerList}>
        <>
          {isLoading && <SpecialistCategoryListShimmer />}
          {!isLoading && isFetched() && (
            <SpecialistCategoryList
              specialistsCategories={specialistsCategories}
              onSelectCategory={goToSpecialistListScreen}
            />
          )}
          {!isLoading && !isFetched() && isError && (
            <ErrorContainer
              iconName={isConnected() ? 'close-circle-outline' : 'earth'}
              message={isConnected() ? 'msg.genericError' : 'msg.networkError'}
              onPressTryAgain={() => getAllSpecialistCategory()}
            />
          )}
        </>
      </View>
      <Text style={styles.subtitle} bold>
        {systemI18n.t('title.whatDoYouNeed')}
      </Text>
      <HomeMenuContainer />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    padding: 15,
    paddingTop: 50,
    borderWidth: 0,
    backgroundColor: systemColors.white,
  },
  containerList: {
    paddingTop: 20,
    borderWidth: 0,
    height: systemLayout.window.height / 4,
  },
  welcome: {
    marginTop: 5,
    fontSize: 18,
  },
  username: {
    fontSize: 20,
    fontWeight: 'bold',
    marginBottom: 20,
  },
  subtitle: {
    fontSize: 16,
    fontWeight: 'bold',
    marginTop: 25,
  },
  errorContainer: {
    width: '100%',
    height: systemLayout.window.height / 8,
  },
  errorMsg: {
    borderWidth: 1,
  },
});
