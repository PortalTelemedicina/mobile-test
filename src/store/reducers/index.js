import { combineReducers } from 'redux'
import specialist from './specialist'

const reducers = combineReducers({
  specialist
})

export default reducers
