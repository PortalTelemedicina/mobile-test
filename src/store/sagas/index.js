import { all } from 'redux-saga/effects'
import specialistSaga from './specialist'

export default function* rootSaga() {
  yield all([specialistSaga()])
}
