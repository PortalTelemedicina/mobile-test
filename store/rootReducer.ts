import { combineReducers } from 'redux';

import specialistCategoryReducer from './specialistCategory/specialistCategoryReducer';
import specialistReducer from './specialist/specialistReducer';
import systemReducer from './system/systemReducer';

const rootReducer = combineReducers({
  specialistCategory: specialistCategoryReducer,
  specialist: specialistReducer,
  system: systemReducer,
});

export type AppState = ReturnType<typeof rootReducer>;

export default rootReducer;
