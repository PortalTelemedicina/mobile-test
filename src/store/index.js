import { createStore, compose, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';

import reducers from './reducers';
import sagas from './sagas';

const middlewares = [];

const sagaMiddleware = createSagaMiddleware({ });

middlewares.push(sagaMiddleware);

const composer = __DEV__
  ? compose(
    applyMiddleware(...middlewares)
  )
  : compose(applyMiddleware(...middlewares));

const store = createStore(reducers, composer);

sagaMiddleware.run(sagas);

export default store;
