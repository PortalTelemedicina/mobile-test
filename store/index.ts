import { createStore, applyMiddleware } from 'redux';
import { persistReducer, persistStore } from 'redux-persist';
import AsyncStorage from '@react-native-community/async-storage';
import createSagaMiddleware from 'redux-saga';
// import logger from 'redux-logger';

import rootReducer from './rootReducer';
import { rootSaga } from './rootSaga';

const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
  whitelist: ['specialistCategory', 'specialist'],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const sagaMiddleware = createSagaMiddleware();

const store = createStore(persistedReducer, applyMiddleware(sagaMiddleware));

sagaMiddleware.run(rootSaga);

const persistor = persistStore(store);

export { store, persistor };
