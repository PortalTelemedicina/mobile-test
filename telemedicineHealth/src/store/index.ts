import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';

import { ISpecialistsState } from './modules/specialists/types';
import { ISpecialistTypeState } from './modules/listTypeSpecialist/types';
import rootReducer from './modules/rootReducer';
import rootSaga from './modules/rootSaga';

export interface IState {
  specialist: ISpecialistsState;
  typeSpecialist: ISpecialistTypeState;
}

const sagaMiddleware = createSagaMiddleware();
const storeEnhancer = applyMiddleware(sagaMiddleware);
const enhancedCreateStore = storeEnhancer(createStore);
const store = enhancedCreateStore(rootReducer);
sagaMiddleware.run(rootSaga);

export default store;
