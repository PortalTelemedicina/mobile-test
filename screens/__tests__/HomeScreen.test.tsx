/* eslint-disable @typescript-eslint/ban-ts-comment */
/* eslint-disable import/extensions */
/* eslint-disable global-require */
/* eslint-disable @typescript-eslint/no-var-requires */
import React from 'react';
import axios from 'axios';
import renderer, { act } from 'react-test-renderer';
import { Provider } from 'react-redux';
import HomeScreen from '../HomeScreen';
import specislistCategories from '../../docs/api/home_specialists.json';
import store from '../../helpers/TestUtils';

function renderWithProviders(ui: JSX.Element) {
  return renderer.create(<Provider store={store}>{ui}</Provider>);
}

jest.mock('axios');
const mockedAxios = axios as jest.Mocked<typeof axios>;

jest.mock('@expo/vector-icons', () => {
  const { View } = require('react-native');
  return {
    MaterialCommunityIcons: View,
  };
});

jest.useFakeTimers();

describe('HomeScreen', () => {
  it('Load specilistCatgegoryList', async () => {
    mockedAxios.get.mockResolvedValue({
      data: specislistCategories,
    });

    let component;
    await act(async () => {
      // @ts-ignore
      component = renderWithProviders(<HomeScreen />);
    });
    // @ts-ignore: Object is possibly 'null'.
    const testInstance = component?.root;
    expect(testInstance.findByProps({ testID: 'specialist-category-0' })).toBeDefined();
    expect(testInstance.findByProps({ testID: 'specialist-category-1' })).toBeDefined();
    expect(testInstance.findByProps({ testID: 'specialist-category-2' })).toBeDefined();
    // @ts-ignore: Object is possibly 'null'.
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Internet error', async () => {
    mockedAxios.get.mockRejectedValueOnce(new Error('Network Error'));
    let component;
    await act(async () => {
      // @ts-ignore
      component = renderWithProviders(<HomeScreen />);
    });

    // @ts-ignore: Object is possibly 'null'.
    const testInstance = component?.root;
    expect(testInstance.findByProps({ testID: 'msg.networkError' })).toBeDefined();

    // @ts-ignore: Object is possibly 'null'.
    // expect(component.toJSON()).toMatchSnapshot();
  });
});
