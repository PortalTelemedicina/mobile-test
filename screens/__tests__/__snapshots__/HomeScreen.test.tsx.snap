// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen Load specilistCatgegoryList 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderWidth": 0,
      "flex": 1,
      "flexDirection": "column",
      "padding": 15,
      "paddingTop": 50,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#504c4c",
          "fontFamily": "ubuntu",
        },
        Object {
          "fontSize": 18,
          "marginTop": 5,
        },
      ]
    }
  >
    Hello,
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#504c4c",
          "fontFamily": "ubuntu-bold",
        },
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 20,
        },
      ]
    }
  >
    Lorelle Lunna
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#504c4c",
          "fontFamily": "ubuntu-bold",
        },
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
          "marginTop": 25,
        },
      ]
    }
  >
    Specialists
  </Text>
  <View
    style={
      Object {
        "borderWidth": 0,
        "height": 333.5,
        "paddingTop": 20,
      }
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "color": "#E5495E",
            "image_url": "https://raw.githubusercontent.com/PortalTelemedicina/mobile-test/main/icons/heart-shape-outline-with-lifeline.svg",
            "name": "Heart Specialist",
            "total": 10,
          },
          Object {
            "color": "#F6AF3D",
            "image_url": "https://raw.githubusercontent.com/PortalTelemedicina/mobile-test/main/icons/tooth.svg",
            "name": "Dental Care",
            "total": 2,
          },
          Object {
            "color": "#7349E5",
            "image_url": "https://raw.githubusercontent.com/PortalTelemedicina/mobile-test/main/icons/pimples.svg",
            "name": "Dermatology Specialist",
            "total": 4,
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsHorizontalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="specialist-category-0"
          >
            <View
              style={
                Array [
                  Object {
                    "borderRadius": 10,
                    "height": 296.44444444444446,
                    "marginRight": 10,
                    "padding": 10,
                    "width": 250,
                  },
                  Object {
                    "backgroundColor": "#E5495E",
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#fff",
                    "borderRadius": 10,
                    "width": "50%",
                  }
                }
              >
                <View
                  color="#E5495E"
                  name="heart-outline"
                  size={50}
                  style={
                    Object {
                      "textAlign": "center",
                    }
                  }
                />
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#504c4c",
                      "fontFamily": "ubuntu-bold",
                    },
                    Object {
                      "color": "#fff",
                      "fontSize": 16,
                      "paddingTop": 5,
                    },
                  ]
                }
              >
                Heart Specialist
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#504c4c",
                      "fontFamily": "ubuntu",
                    },
                    Object {
                      "color": "#fff",
                      "fontSize": 14,
                      "paddingTop": 5,
                    },
                  ]
                }
              >
                10 Doctors
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="specialist-category-1"
          >
            <View
              style={
                Array [
                  Object {
                    "borderRadius": 10,
                    "height": 296.44444444444446,
                    "marginRight": 10,
                    "padding": 10,
                    "width": 250,
                  },
                  Object {
                    "backgroundColor": "#F6AF3D",
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#fff",
                    "borderRadius": 10,
                    "width": "50%",
                  }
                }
              >
                <View
                  color="#F6AF3D"
                  name="tooth-outline"
                  size={50}
                  style={
                    Object {
                      "textAlign": "center",
                    }
                  }
                />
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#504c4c",
                      "fontFamily": "ubuntu-bold",
                    },
                    Object {
                      "color": "#fff",
                      "fontSize": 16,
                      "paddingTop": 5,
                    },
                  ]
                }
              >
                Dental Care
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#504c4c",
                      "fontFamily": "ubuntu",
                    },
                    Object {
                      "color": "#fff",
                      "fontSize": 14,
                      "paddingTop": 5,
                    },
                  ]
                }
              >
                2 Doctors
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="specialist-category-2"
          >
            <View
              style={
                Array [
                  Object {
                    "borderRadius": 10,
                    "height": 296.44444444444446,
                    "marginRight": 10,
                    "padding": 10,
                    "width": 250,
                  },
                  Object {
                    "backgroundColor": "#7349E5",
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#fff",
                    "borderRadius": 10,
                    "width": "50%",
                  }
                }
              >
                <View
                  color="#7349E5"
                  name="head-outline"
                  size={50}
                  style={
                    Object {
                      "textAlign": "center",
                    }
                  }
                />
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#504c4c",
                      "fontFamily": "ubuntu-bold",
                    },
                    Object {
                      "color": "#fff",
                      "fontSize": 16,
                      "paddingTop": 5,
                    },
                  ]
                }
              >
                Dermatology Specialist
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#504c4c",
                      "fontFamily": "ubuntu",
                    },
                    Object {
                      "color": "#fff",
                      "fontSize": 14,
                      "paddingTop": 5,
                    },
                  ]
                }
              >
                4 Doctors
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#504c4c",
          "fontFamily": "ubuntu-bold",
        },
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
          "marginTop": 25,
        },
      ]
    }
  >
    What do you need?
  </Text>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "space-between",
        "paddingTop": 20,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "aspectRatio": 1,
            "backgroundColor": "#f2f2f2",
            "borderRadius": 10,
            "height": 214.28571428571428,
            "justifyContent": "center",
            "margin": "1%",
            "padding": 10,
            "textAlign": "center",
            "width": 214.28571428571428,
          },
          Object {
            "backgroundColor": "#504c4c",
          },
        ]
      }
    >
      <View>
        <View
          color="#f2f2f2"
          name="pulse"
          size={50}
          style={
            Object {
              "textAlign": "center",
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#504c4c",
              "fontFamily": "ubuntu",
            },
            Object {
              "color": "#f2f2f2",
              "textAlign": "center",
            },
          ]
        }
      >
        Diagnostic
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "aspectRatio": 1,
            "backgroundColor": "#f2f2f2",
            "borderRadius": 10,
            "height": 214.28571428571428,
            "justifyContent": "center",
            "margin": "1%",
            "padding": 10,
            "textAlign": "center",
            "width": 214.28571428571428,
          },
          null,
        ]
      }
    >
      <View>
        <View
          color="#504c4c"
          name="account-arrow-left-outline"
          size={50}
          style={
            Object {
              "textAlign": "center",
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#504c4c",
              "fontFamily": "ubuntu",
            },
            Object {
              "color": "#504c4c",
              "textAlign": "center",
            },
          ]
        }
      >
        Consultation
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "aspectRatio": 1,
            "backgroundColor": "#f2f2f2",
            "borderRadius": 10,
            "height": 214.28571428571428,
            "justifyContent": "center",
            "margin": "1%",
            "padding": 10,
            "textAlign": "center",
            "width": 214.28571428571428,
          },
          null,
        ]
      }
    >
      <View>
        <View
          color="#504c4c"
          name="doctor"
          size={50}
          style={
            Object {
              "textAlign": "center",
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#504c4c",
              "fontFamily": "ubuntu",
            },
            Object {
              "color": "#504c4c",
              "textAlign": "center",
            },
          ]
        }
      >
        Nurse
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "aspectRatio": 1,
            "backgroundColor": "#f2f2f2",
            "borderRadius": 10,
            "height": 214.28571428571428,
            "justifyContent": "center",
            "margin": "1%",
            "padding": 10,
            "textAlign": "center",
            "width": 214.28571428571428,
          },
          null,
        ]
      }
    >
      <View>
        <View
          color="#504c4c"
          name="ambulance"
          size={50}
          style={
            Object {
              "textAlign": "center",
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#504c4c",
              "fontFamily": "ubuntu",
            },
            Object {
              "color": "#504c4c",
              "textAlign": "center",
            },
          ]
        }
      >
        Ambulance
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "aspectRatio": 1,
            "backgroundColor": "#f2f2f2",
            "borderRadius": 10,
            "height": 214.28571428571428,
            "justifyContent": "center",
            "margin": "1%",
            "padding": 10,
            "textAlign": "center",
            "width": 214.28571428571428,
          },
          null,
        ]
      }
    >
      <View>
        <View
          color="#504c4c"
          name="flask-outline"
          size={50}
          style={
            Object {
              "textAlign": "center",
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#504c4c",
              "fontFamily": "ubuntu",
            },
            Object {
              "color": "#504c4c",
              "textAlign": "center",
            },
          ]
        }
      >
        Lab Work
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "aspectRatio": 1,
            "backgroundColor": "#f2f2f2",
            "borderRadius": 10,
            "height": 214.28571428571428,
            "justifyContent": "center",
            "margin": "1%",
            "padding": 10,
            "textAlign": "center",
            "width": 214.28571428571428,
          },
          null,
        ]
      }
    >
      <View>
        <View
          color="#504c4c"
          name="bottle-tonic-plus-outline"
          size={50}
          style={
            Object {
              "textAlign": "center",
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#504c4c",
              "fontFamily": "ubuntu",
            },
            Object {
              "color": "#504c4c",
              "textAlign": "center",
            },
          ]
        }
      >
        Medicine
      </Text>
    </View>
  </View>
</View>
`;
